python/eight_mile/pytorch/layers.py

```diff
@@ -1718,6 +1720,91 @@ def _attention(self, query: torch.Tensor, key: torch.Tensor, mask: Optional[torc
         return F.softmax(scores, dim=-1)
 
 
+class SequenceSequenceRelativeAttention(nn.Module):
+    """This form of attention is specified in Shaw et al 2018: https://www.aclweb.org/anthology/N18-2074.pdf
+
+    TODO: where to form RPR?
+    """
+    def __init__(self, hsz: int = None, pdrop: float = 0.1, **kwargs):
+        super().__init__()
+        self.hsz = hsz
+        self.dropout = nn.Dropout(pdrop)
+        self.attn = None
+
+    def forward(self, qkvrm: Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]) -> torch.Tensor:
+        query, key, value, edges_key, edges_value, mask = qkvrm
```

Comment:
The name `qkvrm` and the type hint suggests there are 5 things in the tuple but this unpacks into 6 elements

Reply:
The type hints will show this but I think probably commenting is the best solution.  We could come up with some delimiters in the name, e.g. `q_k_v_rr_rv_m`.

Reply:
There are only 5 `torch.Tensor`s in the `Tuple` in the type hint.